### PR TITLE
#2928 - Updated all error messages to have error icon - [AB]

### LINKED ISSUE #2928: Bundle: Error messages and alert accessibility
### Story

As a assistive technology enabled user
I want errors preventing me from moving forward to be clearly called out
so that I can fix the error without confusion

Our error alerts and fields fail the following WCAG criteria today:
- [4.1.2 Name, Role, Value:](https://www.w3.org/TR/WCAG20/#ensure-compat-rsv)
- [1.4.1 Use of Color](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color)

### Acceptance Criteria

- [ ] #2943
- [ ] #2964
- [ ] Be sure to check all pages, including the review page on request, and check using both the org model flag an non org model flag.

### Additional Context

This story resolves issue 7 and 8 from the Trusted Tester 508 report and [remediation plan](https://docs.google.com/document/d/1L7TW2RdxAFUrWWLSw3IUYcy4Zu_1OnBl/edit).

### Issue Links
- Relates to #2913 

### Prioritization Comments
**WCAG Conformance Level:** 
**WCAG Standard that applies:** 
**Critical Failure if Not Met?:**  
**Prioritization:** 
**Reason:**

### PR DESCRIPTION
**Pull request by CocoByte**
## Ticket 2928

Resolves #2928

## Changes

- Updated all error messages to include an error icon.  The icon has a screenreader output that says "Errror"

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Update documentation in READMEs and/or onboarding guide

#### Ensured code standards are met (Original Developer)

- [ ] If any updated dependencies on Pipfile, also update dependencies in requirements.txt.
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [ ] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [x] Checked behavior. Comment any found issues or broken flows.
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

### CODE REVIEW

**Review by SamiyahKey — COMMENTED** [review]
looks good! ANDI output lets me know that I have and invalid entry and to correct the specified field, and the error icon is identified

File: src/registrar/templates/includes/input_with_errors.html [hunk 1]
```diff
@@ -52,9 +52,14 @@
   {% if field.errors %}
     <div id="{{ widget.attrs.id }}__error-message">
     {% for error in field.errors %}
-      <span class="usa-error-message" role="alert">
-        {{ error }}
-      </span>
+      <div class="usa-error-message display-flex" role="alert">
+        <div class="margin-top-05">
+          <svg class="usa-icon usa-icon-large" focusable="true" role="img" aria-label="Error">
```
Comment (rachidatecs) [inline src/registrar/templates/includes/input_with_errors.html:57]:
```suggestion
          <svg class="usa-icon usa-icon--large" focusable="true" role="img" aria-label="Error">
```

File: src/registrar/assets/sass/_theme/_alerts.scss [hunk 2]
```diff
@@ -51,3 +51,8 @@
 .usa-site-alert--hot-pink .usa-alert .usa-alert__body::before {
   background-image: url('../img/usa-icons-bg/error.svg');
 }
+
+.usa-icon-large {
```
Comment (rachidatecs) [inline src/registrar/assets/sass/_theme/_alerts.scss:55]:
```suggestion
.usa-icon--large {
```

**Comment by rachidatecs** [comment]
@CocoByte, see the existing `usa-icon--big` class used on buttons and see if you can repurpose this instead of adding a redundant class.

Only extra line of style on `usa-icon--big` is `margin: 0; `which I think we can get away with. We'd also want to move out the block from `_buttons.scss `to `_base.scss` and put in a comment that this class is used by buttons and form errors.

**Review by rachidatecs — CHANGES_REQUESTED** [review]
Ignore my first review and see the freeform comments. We have an existing class we can use for this. I'd be happy if you renamed the consolidated class usa-button--large instead of the --big

File: src/registrar/templates/includes/input_with_errors.html [hunk 3]
```diff
@@ -52,9 +52,14 @@
   {% if field.errors %}
     <div id="{{ widget.attrs.id }}__error-message">
     {% for error in field.errors %}
-      <span class="usa-error-message" role="alert">
-        {{ error }}
-      </span>
+      <div class="usa-error-message display-flex" role="alert">
+        <div>
```
Comment (rachidatecs) [inline src/registrar/templates/includes/input_with_errors.html:56]: Is the extra div wrapper needed here?

Comment (CocoByte) [reply src/registrar/templates/includes/input_with_errors.html:56]: Nope.  Remnant from former cleanup.  Good catch

**Review by rachidatecs — APPROVED** [review]

**Pull request merged**